### DISCUSSION TITLE
Error installing pyodbc

### DISCUSSION THREAD

**Discussion by HernaniMourao** [tiangolo/uwsgi-nginx-flask-docker#363] [Questions] [answered]
Hello,
I am unable to install pyodbc on a recently created image.

Appreciate any help on the subject...
Regards,
Hernâni

**Comment by HernaniMourao** [comment]
I should add:

  Building wheel for pyodbc (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-aguml8hj/pyodbc_619fcc1699ec4c76ba33bdc1e822d2f5/setup.py'"'"'; __file__='"'"'/tmp/pip-install-aguml8hj/pyodbc_619fcc1699ec4c76ba33bdc1e822d2f5/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-rld1wbmv
       cwd: /tmp/pip-install-aguml8hj/pyodbc_619fcc1699ec4c76ba33bdc1e822d2f5/
  Complete output (14 lines):
  running bdist_wheel
  running build
  running build_ext
  building 'pyodbc' extension

**Comment by tiangolo** [comment, marked as answer]
It would be better if you provide your custom `Dockerfile`, make sure you are not using a slim image as it seems you need to compile some C extensions for it to work. Also make sure you have a recent version of `pip`, maybe there's already a pre-compiled version of pyodbc and you just need to have a recent `pip` to be able to install it.
